torch_geometric/nn/conv/rgat_conv.py

```diff
@@ -0,0 +1,558 @@
+            self.bias = Parameter(
+                torch.Tensor(self.heads * self.d * self.out_channels))
+        elif bias and not concat:
+            self.bias = Parameter(torch.Tensor(self.d * self.out_channels))
+        else:
+            self.register_parameter('bias', None)
+
+        if edge_dim is not None:
+            self.lin_edge = Linear(self.edge_dim,
+                                   self.heads * self.out_channels, bias=False,
+                                   weight_initializer='glorot')
+            self.e = Parameter(
+                torch.Tensor(self.heads * self.out_channels,
+                             self.heads * self.d))
+        else:
+            self.lin_edge = None
+            self.register_parameter('e', None)
+
+        if num_bases is not None:
+            self.att = Parameter(
+                torch.Tensor(self.num_relations, self.num_bases))
+            self.basis = Parameter(
+                torch.Tensor(self.num_bases, self.in_channels,
+                             self.heads * self.out_channels))
+        elif num_blocks is not None:
+            assert (self.in_channels % self.num_blocks == 0
+                    and (self.heads * self.out_channels) %
+                    self.num_blocks == 0), \
+                "both 'in_channels' and 'heads * out_channels' must be " \
+                "multiple of 'num_blocks' used"
+            self.weight = Parameter(
+                torch.Tensor(self.num_relations, self.num_blocks,
+                             self.in_channels // self.num_blocks,
+                             (self.heads * self.out_channels) //
+                             self.num_blocks))
+        else:
+            self.weight = Parameter(
+                torch.Tensor(self.num_relations, self.in_channels,
+                             self.heads * self.out_channels))
+
+        self.w = Parameter(torch.ones(self.out_channels))
+        self.l1 = Parameter(torch.FloatTensor(1, self.out_channels))
+        self.b1 = Parameter(torch.FloatTensor(1, self.out_channels))
+        self.l2 = Parameter(
+            torch.FloatTensor(self.out_channels, self.out_channels))
+        self.b2 = Parameter(torch.FloatTensor(1, self.out_channels))
+
+        self._alpha = None
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        if self.num_bases is not None:
+            glorot(self.basis)
+            glorot(self.att)
+        else:
+            glorot(self.weight)
+        glorot(self.q)
+        glorot(self.k)
+        zeros(self.bias)
+        ones(self.l1)
+        zeros(self.b1)
+        torch.full(self.l2.size(), 1 / self.out_channels)
+        zeros(self.b2)
+        if self.lin_edge is not None:
+            glorot(self.lin_edge)
+            glorot(self.e)
+
+    def forward(self, x: Tensor, edge_index: Adj, edge_type: OptTensor = None,
+                edge_attr: OptTensor = None, size: Size = None,
+                return_attention_weights=None):
```

Comment:
```suggestion
                return_attention_weights: bool = False):
```

Reply:
This line (attention weights will be returned whenever "return_attention_weights" is set to a value, regardless of its actual value (might be `True` or `False`)) is taken from [https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/gat_conv.py](url). Hence, I've to set "return_attention_weights" to None in order to allow the layer not to return any attention weight (by default).